src/mission_ui.cpp

```diff
@@ -451,20 +454,27 @@ void mission_ui_impl::draw_selected_description( std::vector<point_of_interest>
     // Avoid replacing expanded snippets with other valid snippet text on redraw
     cataimgui::draw_colored_text( parsed_description, c_unset, table_column_width * 1.15 );
     // TODO: target does not contain a z-component, targets are assumed to be on z=0
+    draw_location( _( "Target:" ), selected_point_of_interest.pos );
+}
+
+void mission_ui_impl::draw_location( const std::string &label,
+                                     const tripoint_abs_omt &loc ) const
+{
     const tripoint_abs_omt pos = get_player_character().pos_abs_omt();
-    cataimgui::draw_colored_text( string_format( _( "Target: %s" ),
-                                  selected_point_of_interest.pos.to_string() ),
-                                  c_white );
-    // Below is done instead of a table for the benefit of right-to-left languages
-    //~Extra padding spaces in the English text are so that the replaced string vertically aligns with the one above
-    cataimgui::draw_colored_text( string_format( _( "You:    %s" ), pos.to_string() ), c_white );
-    int omt_distance = rl_dist( pos, selected_point_of_interest.pos );
+    draw_label_with_value( label, display::overmap_position_text( loc ) );
+    if( !you_know_where_you_are() ) {
+        // Don't display "Distance:" or direction arrow if we don't know where we are
+        return;
+    }
+    int omt_distance = rl_dist( pos, loc );
     if( omt_distance > 0 ) {
         // One OMT is 24 tiles across, at 1x1 meters each, so we can simply do number of OMTs * 24
         units::length actual_distance = omt_distance * 24_meter;
-        //~Parenthesis is a real-world value for distance. Example string: "Distance: 223 tiles (5352 m)"
-        cataimgui::draw_colored_text( string_format( _( "Distance: %1$s tiles (%2$s)" ),
-                                      omt_distance, length_to_string_approx( actual_distance ) ), c_white );
+        const std::string dir_arrow = direction_arrow( direction_from( pos.xy(), loc.xy() ) );
+        //~Parenthesis is a real-world value for distance. Example string: "223 tiles (5.35km) ⇗"
+        const std::string distance_str = string_format( _( "%1$s tiles (%2$s) %3$s" ),
+                                         omt_distance, length_to_string_approx( actual_distance ), dir_arrow );
```

Comment:
Hmm. It would be better not to create a temporary string here, since this happens every frame. However, the code was already doing it that way so I won’t make you refactor it. We can do that later in a separate pull request. Or maybe we can make Renech do it, since he wrote this code.